components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java

```diff
@@ -642,9 +642,14 @@ protected String encodeRequestMessage(RequestAbstractType requestMessage, String
             for (AttributeStatement statement : attributeStatementList) {
                 List<Attribute> attributesList = statement.getAttributes();
                 for (Attribute attribute : attributesList) {
-                    Element value = attribute.getAttributeValues().get(0).getDOM();
-                    String attributeValue = value.getTextContent();
-                    results.put(attribute.getName(), attributeValue);
+                    StringBuilder attributeValue = new StringBuilder("");
```

Comment:
Just a suggestion. We can put the attribute values to a list and use a one liner to join them with a delimiter

Ref: https://stackoverflow.com/questions/4021851/join-string-list-elements-with-a-delimiter-in-one-step